sdk/python/tests/compiler/compiler_tests.py

```diff
@@ -1003,3 +1003,8 @@ def some_pipeline():
         self.assertNotIn(' ', argument['name'], 'The input name "{}" of template "{}" was not sanitized.'.format(argument['name'], template['name']))
       for argument in template['inputs']['artifacts']:
         self.assertNotIn(' ', argument['name'], 'The input name "{}" of template "{}" was not sanitized.'.format(argument['name'], template['name']))
+
+  def test_keyword_only_argument_for_pipeline_func(self):
+    def some_pipeline(casual_argument: str, *, keyword_only_argument: str):
+      pass
+    kfp.compiler.Compiler().compile(some_pipeline)
```

Comment:
You can pass None as path.

Reply:
Oh, right! I saw `_compile` in tests but it was marked as "deprecated, use compile", so I just mindlessly replaced it, I guess...

Reply:
`_compile` is fine as well. We might de-deprecate it... The deprecation was due to a way TFX SDK was using the private compiler functions.